home/service/glossary.py

```diff
@@ -0,0 +1,46 @@
+from data_platform_catalogue.search_types import MultiSelectFilter, ResultType
+from django.core.exceptions import ObjectDoesNotExist
+from copy import deepcopy
+from itertools import groupby
+
+from .base import GenericService
+
+
+class GlossaryService(GenericService):
+    def __init__(self):
+        # Can we put the client instantiation in the base class?
+        self.client = self._get_catalogue_client()
+        self.context = self._get_context()
+
+    def _get_context(self):
+        """Returns a glossary context which is grouped by parent term"""
+        glossary_search_results = self.client.get_glossary_terms()
+        total_results = glossary_search_results.total_results
+
+        # The sort is required for the grouping to work correctly
+        page_results_copy = deepcopy(glossary_search_results.page_results)
```

Comment:
If you use `sorted(list)` rather than `list.sort` it will create a copy as part of the sort, so you might not need this line(?)

Reply:
I will apply this